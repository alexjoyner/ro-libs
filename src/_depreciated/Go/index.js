import React from 'react';
import styled from 'styled-components';

const Template = styled.a`
    text-decoration: none;
    color: inherit;
`;

export const Go = ({ to, children, ...props }) => (
  <Template href={to} {...props}>
    {children}
  </Template>
);
