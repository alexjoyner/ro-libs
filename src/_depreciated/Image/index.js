import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './particles/Container';
import { Pic } from './particles/Pic';


const Image = ({ ...props }) => (
  <Container>
    <Pic {...props} />
  </Container>
);

Image.propTypes = {
  imgStyle: PropTypes.string,
  imgSize: PropTypes.string,
};
Image.defaultProps = {
  imgStyle: 'default',
  imgSize: 'default',
};

export { Image };
