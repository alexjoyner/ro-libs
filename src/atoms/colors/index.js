export const colors = {
  primaryLight: '#74b9ff',
  primary: '#0984e3',
  successLight: '#55efc4',
  success: '#00b894',
  warningLight: '#ffeaa7',
  warning: '#fdcb6e',
  dangerLight: '#ff7675',
  danger: '#d63031',
};
