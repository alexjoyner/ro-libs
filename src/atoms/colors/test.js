import { colors } from './';

describe('colors', () => {
  it('Should match all default colors', () => {
    expect(colors).toMatchSnapshot();
  });
});
