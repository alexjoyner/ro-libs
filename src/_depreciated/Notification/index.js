import React from 'react';
import styled from 'styled-components';
import { IoMdClose } from 'react-icons/io';
import { Panel } from '../../atoms/Panel';
import { CenteredContent } from '../CenteredContent';
import { Button } from '../../atoms/Button';
import { zIndexes } from '../../utils/zIndexes';

const NotificationWrapper = styled.div`
    position: fixed;
    z-index: ${zIndexes.Notification};
    top: 40px;
    right: 15px;
    transition: all 0.5s ease;
    ${props => ((props.shown) ? `
        transform: translateX(0)
    ` : `
        transform: translateX(500%)
    `)}
`;


const Notification = ({ children, onClose, ...props }) => (
  <NotificationWrapper {...props}>
    <Panel width="300px">
      <CenteredContent>
        {children}
        <Button color="danger" size="small" onClick={onClose}>
          <IoMdClose />
        </Button>
      </CenteredContent>
    </Panel>
  </NotificationWrapper>
);

export { Notification };
