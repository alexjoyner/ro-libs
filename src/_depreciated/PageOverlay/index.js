import React from 'react';
import { Overlay } from '../../atoms/Overlay';


const PageOverlay = props => <Overlay model="page" {...props} />;

export { PageOverlay };
