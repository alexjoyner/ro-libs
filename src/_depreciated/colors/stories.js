import React from 'react';
import { Atoms } from '../../atoms/_atoms.stories';
import { Panel } from '../../atoms/Panel';
import { colors } from '.';
import { CenteredContent } from '../CenteredContent';
import { getTextContrastColor } from '../../utils/getTextContrastColor';

Atoms
  .add('Colors', () => (
    <div>
      <Panel>
        <CenteredContent>
          <h2>Component library color pallete</h2>
          {Object.keys(colors).map(key => (
            <div style={{
              height: '100px',
              width: '100%',
              backgroundColor: colors[key],
              color: getTextContrastColor(colors[key]),
            }}
            >
              <span>{key}</span>
            </div>
          ))}
        </CenteredContent>
      </Panel>
    </div>
  ));
