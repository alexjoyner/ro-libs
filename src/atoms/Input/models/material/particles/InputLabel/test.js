import React from 'react';
import { mount } from 'enzyme';
import { InputLabel } from '.';
import { colors } from '../../../../../../_depreciated/colors';

describe('InputLabel component', () => {
  let component;
  beforeEach(() => {
    component = mount(<InputLabel inputValue="" />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have correct default color', () => {
    expect(component).toHaveStyleRule('color', colors.primary);
  });
  it('should have correct color when error', () => {
    const errTest = mount(<InputLabel error />);
    expect(errTest).toHaveStyleRule('color', colors.danger);
  });
  it('should have correct default positioning', () => {
    expect(component).toHaveStyleRule('top', '50%');
  });
  it('should have correct positioning when input value changes', () => {
    const valChange = mount(<InputLabel inputValue="Testing" />);
    expect(valChange).toHaveStyleRule('top', '2px');
    expect(valChange).toHaveStyleRule('font-size', '12px');
  });
});
