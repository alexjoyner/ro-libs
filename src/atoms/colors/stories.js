// @flow
import React from 'react';
import { Atoms } from '../_atoms.stories';
import { Panel } from '../../atoms/Panel';
import { colors } from './';
import { CenteredContent } from '../../atoms/CenteredContent';
import { getTextContrastColor } from '../../utils/getTextContrastColor';

type Props = {
  color: string,
  name: string,
}
const TestColorBox = (props: Props) => (
  <div style={{
    height: '100px',
    width: '100%',
    backgroundColor: props.color,
    color: getTextContrastColor(props.color),
  }}
  >
    <span>{props.name}</span>
  </div>
);

Atoms
  .add('Colors', () => (
    <div>
      <Panel>
        <CenteredContent>
          <h2>Component library color pallete</h2>
          {Object.keys(colors).map(key => (<TestColorBox
            key={key}
            color={colors[key]}
            name={key}
          />))}
        </CenteredContent>
      </Panel>
    </div>
  ));

