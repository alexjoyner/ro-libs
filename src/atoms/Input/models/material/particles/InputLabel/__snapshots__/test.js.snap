// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel component Should match snapshot 1`] = `
.c0 {
  position: absolute;
  left: 0px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
  -webkit-transition: 50ms linear;
  transition: 50ms linear;
  color: #2980b9;
  top: 50%;
}

<InputLabel
  inputValue=""
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "InputLabel-hk3wpt-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "position:absolute;left:0px;transform:translateY(-50%);pointer-events:none;transition:50ms linear;color:",
            [Function],
            ";",
            [Function],
          ],
        },
        "displayName": "InputLabel",
        "render": [Function],
        "styledComponentId": "InputLabel-hk3wpt-0",
        "target": "label",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    inputValue=""
  >
    <label
      className="c0"
    />
  </StyledComponent>
</InputLabel>
`;
